bundle-workflow/python/sign.py

```diff
@@ -1,37 +1,31 @@
 #!/usr/bin/env python
 
 import os
-import subprocess as sp
-import tempfile
-import uuid
 import argparse
-from manifests.bundle_manifest import BundleManifest
+from git.git_repository import GitRepository
+from manifests.build_manifest import BuildManifest
+from signing_workflow.signer import Signer
 
 parser = argparse.ArgumentParser(description = "Sign artifacts")
-parser.add_argument('manifest', type = argparse.FileType('r'), help="Manifest file.")
+parser.add_argument('manifest', type = argparse.FileType('r'), help="Path to local manifest file.")
+parser.add_argument('--component',nargs='?',help="Component name")
+parser.add_argument('--type',nargs='?',help="Artifact type")
 args = parser.parse_args()
 
-def sign(filename):
-    signatureFile = filename + ".asc"
-    signing_cmd = './opensearch-signer-client -i ' + filename + ' -o ' + signatureFile + ' -p pgp'
-    sp.run(signing_cmd.split(),cwd='opensearch-signer-client/src')
-    verify_cmd = 'gpg --verify-files ' + signatureFile
-    sp.run(verify_cmd.split(),cwd='opensearch-signer-client/src')
-    sp.run(['mv',signatureFile,'signed_artifacts/'])
+manifest = BuildManifest.from_file(args.manifest)
+basepath = os.path.dirname(os.path.abspath(args.manifest.name))
+git_repo = GitRepository("https://github.com/opensearch-project/opensearch-signer-client.git", "HEAD")
+signer = Signer(git_repo)
 
-manifest = BundleManifest.from_file(args.manifest)
-output_dir = os.path.join(os.getcwd(), 'signed_artifacts')
-os.makedirs(output_dir, exist_ok = True)
-build_id = os.getenv('OPENSEARCH_BUILD_ID', uuid.uuid4().hex)
-
-os.system('git clone https://github.com/opensearch-project/opensearch-signer-client.git')
-os.system('cp ./config.cfg ./opensearch-signer-client/src/')
-sp.run(['./bootstrap'],cwd='opensearch-signer-client/src')
-
-print(f'Signing {manifest.build.name} into {output_dir}')
 for component in manifest.components:
+    if (not args.component) or ( args.component and component.name == args.component):
```

Comment:
Probably a bit clearer using the opposite condition and use `continue`. 

Like [here](https://github.com/opensearch-project/opensearch-build/blob/main/bundle-workflow/python/build.py#L36)